### PR TITLE
Prioritize oclc over isbn for physical copy searches

### PR DESCRIPTION
**Pull request by skylerbunny**
This should address issue #422, by making OCLC number the preferred search parameter to Worldcat when it exists, and falling back to ISBN only when it does not.

### CODE REVIEW

**Comment by dvanduzer** [comment]
I'm a little new to this codebase, so would you mind explaining why you also switched it from $isbn to $isbn_10?

**Comment by skylerbunny** [comment]
The best I can say is that I didn't 'switch' anything in the choices made, only the priorities in the logic.

In one code location, 'isbn' has historically been used, and in another, 'isbn_10'.

**Comment by skylerbunny** [comment]
It appears the longer answer is 'because openlibrary/templates/books/edition-sort.html doesn't define 'isbn' at all. There's some additional intelligence at line 27 in the other file ( https://github.com/skylerbunny/openlibrary/blob/3627abd741f4da502b285026926a779e98f62ff4/openlibrary/macros/databarWork.html#L27 ) that doesn't appear in edition-sort.

**Comment by dvanduzer** [comment]
I think my eyes just tricked me when I read the diff the first time. Nevermind!

**Comment by skylerbunny** [comment]
Okay, I'm taking this a step further based on @dvanduzer's question. Regardless of the fact that the intelligence in databarWork.html isn't present in edition-sort, it *should* be for consistency's sake - so the same results will be returned in both places. I've made what I believe are the changes to do that.

**Comment by skylerbunny** [comment]
That should do it - mirrored logic in both files, OCLC prioritized, and worldcat https everywhere I could find it in the code. Ready for review.

File: openlibrary/templates/type/edition-history/view.html [hunk 1]
```diff
@@ -9,7 +9,7 @@
 $var title: $title
 
 $ lccn = page["lccn"] and page["lccn"][0] and ("http://lccn.loc.gov/" + page["lccn"][0])
```
Comment (LeadSongDog) [inline openlibrary/templates/type/edition-history/view.html]: This too should use https. Although loc will redirect anyhow, it's both an extra bounce and a vulnerability.

Comment (LeadSongDog) [inline openlibrary/templates/type/edition-history/view.html]: This too should use https. Although loc.gov will redirect anyhow, it's both an extra bounce and a vulnerability.

Comment (skylerbunny) [inline openlibrary/templates/type/edition-history/view.html]: Done. Scrubbed the other common root query URLs in OL while I was at it to look for sites that fully redirect to https in openlibrary/plugins/openlibrary/pages/config_edition.page and added as appropriate. I doubt it had been done for some time.

Comment (LeadSongDog) [inline openlibrary/templates/type/edition-history/view.html]: Tks

**Comment by mekarpeles** [comment]
Reviewing; thanks @skylerbunny 
P.S. There will be an Open Library community meeting @ 9:30am PST (https://zoom.us/j/850313890) for anyone interested in participating.

**Comment by mekarpeles** [comment]
These changes look to me. We'll pull this onto https://dev.openlibrary.org and make sure it works.

Can you please provide us with a few test cases (what we should expect to see differently from openlibrary.org)?

It looks like it considers isbn10, asin, and scribd ids and also isbn10 + 13 (instead of isbn10 only)

**Comment by mekarpeles** [comment]
Also, would be great if in the future we can make changes (e.g. http -> https) isolated on their own branches as this will make it easier for us to review and merge them in.

**Comment by skylerbunny** [comment]
Real world example: https://github.com/internetarchive/openlibrary/issues/422#issue-210388095 : Using the Worldcat 'find this book' for this OL entry should search OCLC 952649184 (explicitly provided on the record), not ISBN 1449483488.

**Comment by mekarpeles** [comment]
Tested; LGTM

**Pull request merged**